drivers/sensor/sensor_sim/sensor_sim.c

```diff
@@ -199,89 +285,117 @@ static int sensor_sim_init(const struct device *dev)
+		(chan == SENSOR_CHAN_ACCEL_XYZ) ? (SENSOR_CHAN_ACCEL_X) : chan,
+		(chan == SENSOR_CHAN_ACCEL_XYZ) ? (SENSOR_CHAN_ACCEL_Y) : SENSOR_CHAN_PRIV_START,
+		(chan == SENSOR_CHAN_ACCEL_XYZ) ? (SENSOR_CHAN_ACCEL_Z) : SENSOR_CHAN_PRIV_START
+	};
+
+	uint32_t time = k_uptime_get_32();
+	int err = 0;
+
+	k_mutex_lock(&accel_param_mutex, K_FOREVER);
+
+	for (size_t i = 0; (i < ARRAY_SIZE(chans)) && !err; i++) {
+		if (chans[i] == SENSOR_CHAN_PRIV_START) {
+			break;
+		}
+
+		err = wave_gen_generate_value(time, get_wave_params(chans[i]), out_val + i);
+	}
+
+	k_mutex_unlock(&accel_param_mutex);
+
+	if (err) {
+		LOG_ERR("Cannot generate wave value (err %d)", err);
+	}
+
+	return err;
+}
+
+/**
  * @brief Generates accelerometer data.
  *
- * @param chan Channel to generate data for.
+ * @param[in]	chan	Channel to generate data for.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
  */
 static int generate_accel_data(enum sensor_channel chan)
 {
 	int retval = 0;
-	double max_variation = 20.0;
-	static int static_val_coeff = 1.0;
-
-	if (IS_ENABLED(CONFIG_SENSOR_SIM_ACCEL_SINE)) {
-		switch (chan) {
-		case SENSOR_CHAN_ACCEL_X:
-			accel_samples[0] = generate_sine(base_accel_samples[0],
-								max_variation);
-			break;
-		case SENSOR_CHAN_ACCEL_Y:
-			accel_samples[1] = generate_sine(base_accel_samples[1],
-								max_variation);
-			break;
-		case SENSOR_CHAN_ACCEL_Z:
-			accel_samples[2] = generate_sine(base_accel_samples[2],
-								max_variation);
-			break;
-		case SENSOR_CHAN_ACCEL_XYZ:
-			accel_samples[0] = generate_sine(base_accel_samples[0],
-								max_variation);
-			k_sleep(K_MSEC(1));
-			accel_samples[1] = generate_sine(base_accel_samples[1],
-								max_variation);
-			k_sleep(K_MSEC(1));
-			accel_samples[2] = generate_sine(base_accel_samples[2],
-								max_variation);
-			break;
-		default:
-			retval = -ENOTSUP;
-		}
-	} else if (IS_ENABLED(CONFIG_SENSOR_SIM_ACCEL_TOGGLE)) {
-		switch (chan) {
-		case SENSOR_CHAN_ACCEL_X:
-			accel_samples[0] = static_val_coeff * max_variation;
-			break;
-		case SENSOR_CHAN_ACCEL_Y:
-			accel_samples[1] = static_val_coeff * max_variation;
-			break;
-		case SENSOR_CHAN_ACCEL_Z:
-			accel_samples[2] = static_val_coeff * max_variation;
-			break;
-		case SENSOR_CHAN_ACCEL_XYZ:
-			accel_samples[0] = static_val_coeff * max_variation;
-			accel_samples[1] = static_val_coeff * max_variation;
-			accel_samples[2] = static_val_coeff * max_variation;
-			break;
-		default:
-			retval = -ENOTSUP;
-		}
+	generator_function gen_fn = NULL;
 
-		static_val_coeff *= -1.0;
+	if (IS_ENABLED(CONFIG_SENSOR_SIM_ACCEL_WAVE)) {
+		gen_fn = generate_wave;
+	} else if (IS_ENABLED(CONFIG_SENSOR_SIM_ACCEL_TOGGLE)) {
+		gen_fn = generate_toggle;
 	} else {
-		/* Should not happen. */
-		__ASSERT_NO_MSG(false);
+		return -ENOTSUP;
+	}
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		retval = gen_fn(chan, DEFAULT_CHAN_COUNT, &accel_samples[0]);
```

Comment:
I would explicitly tell `1` here instead of `DEFAULT_CHAN_COUNT`

Reply:
fix if you like

Reply:
Replaced `DEFAULT_CHAN_COUNT` with 1